rd-kt/rd-framework/src/main/kotlin/com/jetbrains/rd/framework/impl/RdTask.kt

```diff
@@ -131,8 +138,13 @@ class EndpointWiredRdTask<TReq, TRes>(
         //we are on endpoint side, so listening for cancellation
         RdReactiveBase.logReceived.trace { "received cancellation" }
         buffer.readVoid() //nothing just a void value
-        result.setIfEmpty(RdTaskResult.Cancelled())
-        def.terminate()
+
+        val success = result.setIfEmpty(RdTaskResult.Cancelled())
+        val wireScheduler = call.wireSchedulerIfBound
+        if (success || wireScheduler == null)
+            def.terminate()
+        else if (lifetime.isAlive)
```

Comment:
I think `executeIfAlive` is not needed here, because there is no resource we have to protect. Moreover, we should not prevent termination of lifetime during queueing, because we already want to terminate lifetime.
`if (lifetime.isAlive)` is just an optimization to avoid useless queueing if lifetime is already terminated